conanfile.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from conans import ConanFile, CMake
+from conans.tools import Git
+import os
+
+class ImguiSFML(ConanFile):
+    """ Imgui-SFML library conan file.
+
+    Options:
+
+    * shared: True or False
+      Whether to build this library shared or static
+
+    * fPIC: True or False
+      Whether or not to use -fPIC option while building
+
+    * imconfig: None or String
+      Use 'None' if you want this library to provide default
+      ImGui's imconfig.h file. Otherwise, provide a path to the
+      imconfig.h file you want to include in the build.
+
+    * imconfig_install_folder: None or String
+      Use 'None' if you want this build to include imconfig.h file,
+      otherwise use anything else.
+    """
+
+    name = 'ImGui-SFML'
+    version = '2.0.1'
+    description = 'ImGui binding for use with SFML'
+    topics = ('conan', 'sfml', 'imgui')
+    url = 'https://github.com/eliasdaler/imgui-sfml'
+    homepage = 'https://github.com/eliasdaler/imgui-sfml'
+    author = 'eliasdaler <eliasdaler@yandex.ru>'
+    build_requires = 'sfml/2.5.1@bincrafters/stable'
+    license = 'MIT'
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'imconfig': 'ANY',
+        'imconfig_install_folder': 'ANY'
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'imconfig': None,
+        'imconfig_install_folder': None
+    }
+    exports_sources = ['*.cpp', '*.h', 'CMakeLists.txt', 'cmake*']
+    exports = 'LICENSE.md'
+    _imgui_dir = 'imgui'
+
+    def source(self):
+        git = Git(folder=self._imgui_dir)
+        git.clone('https://github.com/ocornut/imgui.git', 'v1.69')
+
+    def configure(self):
+        self.options['sfml'].graphics = True
+        self.options['sfml'].window = True
+        self.options['sfml'].shared = True
+
+    def configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['IMGUI_DIR'] = os.path.join(self.source_folder, self._imgui_dir)
+        cmake.definitions['SFML_DIR'] = os.path.join(self.deps_cpp_info['sfml'].lib_paths[0], 'cmake', 'SFML')
+        cmake.definitions['IMGUI_SFML_BUILD_EXAMPLES'] = 'OFF'
+        cmake.definitions['IMGUI_SFML_FIND_SFML'] = 'ON'
```

Comment:
Should be "OFF" if you're specifying "SFML_DIR"

Reply:
I had errors because of doing so. Let me check if this is working now..

Reply:
Yes. If I set "SFML_DIR" and disable "IMGUI_SFML_FIND_SFML", then I get compile errors:

```
../imgui-SFML.h:4:10: fatal error: 'SFML/System/Vector2.hpp' file not found
#include <SFML/System/Vector2.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~
```

And some hints from CMake:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CONAN_CMAKE_POSITION_INDEPENDENT_CODE
    CONAN_COMPILER
    CONAN_COMPILER_VERSION
    CONAN_CXX_FLAGS
    CONAN_C_FLAGS
    CONAN_EXPORTED
    CONAN_IN_LOCAL_CACHE
    CONAN_LIBCXX
    CONAN_SHARED_LINKER_FLAGS
    SFML_DIR
```

Reply:
I see. Are you sure that CMake finds the version (via `find_package` in ImGui-SFML's CMakeLists.txt) that Conan downloads because of dependency? If so, then there's no problem, though I don't see why you need to set 'SFML_DIR' in that case.

Reply:
> Are you sure that CMake finds the version <...> that Conan downloads because of dependency?

Yeah. CMake logs confirm that:

```
-- Found SFML 2.5.1 in ~/.conan/data/sfml/2.5.1/bincrafters/stable/package/ed8c36dd1caf5fec34121c6dc29badcbf3252fab/lib/cmake/SFML
-- Found ImGui v1.69 in ~/.conan/data/ImGui-SFML/2.0.1/user/stable/build/eb6bb74157d96eac87398f1d4b33528edcded47f/imgui
```

---

> I don't see why you need to set 'SFML_DIR' in that case.

Hrm. My bet is that's what `find_package` wants from me. With `IMGUI_SFML_FIND_SFML=ON` and no `SFML_DIR` I got this:

``` 
Could not find a package configuration file provided by "SFML" (requested
version 2.5) with any of the following names:

  SFMLConfig.cmake
  sfml-config.cmake

Add the installation prefix of "SFML" to CMAKE_PREFIX_PATH or set
"SFML_DIR" to a directory containing one of the above files.  If "SFML"
provides a separate development package or SDK, be sure it has been
installed.
```

And the content of aformentioned `SFML_DIR`:

```bash
$ ls ~/.conan/data/sfml/2.5.1/bincrafters/stable/package/ed8c36dd1caf5fec34121c6dc29badcbf3252fab/lib/cmake/SFML
SFMLConfig.cmake  SFMLConfigDependencies.cmake  SFMLConfigVersion.cmake  SFMLSharedTargets-release.cmake  SFMLSharedTargets.cmake
```

Reply:
Oh, now I understand how this works. You're getting pre-build SFML, of course, so it contains SFMLConfig.cmake and you need to point CMake to it. Okay, looks good